lib/src/operation_failure.dart

```diff
@@ -1,11 +1,21 @@
 import 'package:rfc_6902/src/operation.dart';
 
 class OperationFailure implements Exception {
-  OperationFailure(this.operation, {this.reason});
+  const OperationFailure(this.operation, {this.reason});
 
   /// The operation which failed.
   final Operation operation;
 
   /// The nested exception which caused failure.
   final Object? reason;
+
+  @override
+  String toString() {
+    final buffer = StringBuffer('OperationFailure: ');
```

Comment:
Seems a bit overkill. Any reason not to go with something like this?
```suggestion
    final description = 'OperationFailure: Failed to perform `${operation.runtimeType}` at path `${operation.path}`';
    return reason != null ? description + ' - $reason' : description;
```